docs/ref/csrf.txt

```diff
@@ -222,24 +222,34 @@ The CSRF protection is based on the following things:
 1. A CSRF cookie that is set to a random value (a session independent nonce, as
    it is called), which other sites will not have access to.
 
-   This cookie is set by ``CsrfViewMiddleware``.  It is meant to be permanent,
-   but since there is no way to set a cookie that never expires, it is sent with
-   every response that has called ``django.middleware.csrf.get_token()``
-   (the function used internally to retrieve the CSRF token).
+   This cookie is set by ``CsrfViewMiddleware``.  It is sent with every
+   response that has called ``django.middleware.csrf.get_token()`` (the
+   function used internally to retrieve the CSRF token), if it wasn't already
+   set on the request.
+
+   In order to protect against `BREACH`_ attacks, the token is not simply the
+   nonce; a random pad is added to the nonce and used to scramble it.
 
    For security reasons, the value of the CSRF cookie is changed each time a
    user logs in.
 
 2. A hidden form field with the name 'csrfmiddlewaretoken' present in all
-   outgoing POST forms.  The value of this field is the value of the CSRF
-   cookie.
+   outgoing POST forms. The value of this field is, again, the value of the
+   nonce, with a pad which is both added to it and used to scramble it. The pad
+   is regenerated on every call to ``get_token()`` so that the form field
+   value is changed in every such response.
 
    This part is done by the template tag.
 
 3. For all incoming requests that are not using HTTP GET, HEAD, OPTIONS or
    TRACE, a CSRF cookie must be present, and the 'csrfmiddlewaretoken' field
    must be present and correct. If it isn't, the user will get a 403 error.
 
+   When validating the 'csrfmiddlewaretoken' field value, only the nonce,
+   not the full token, is compared with the nonce in the cookie value.
+   This allows the use of ever-changing tokens. While each request may use its
+   own token, the nonce remains common to all.
```

Comment:
I think you mixed up "token" and "nonce" here, as pointed out by @apollo13 earlier.